lib/main.js

```diff
@@ -394,7 +394,7 @@ define(['./errors'], function(errors) {
     // for easy membership testing with the "in" keyword
     function setLike(list, fill /* default true*/) {
         if(fill === undefined) fill = true;
-        var setLike = {};
+        var setLike = Object.create(null);
```

Comment:
> in this case, also the use of null when JS already "passes in" undefined is generally considered making it less JS

sorry, but https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/create

> #### Throws
> Throws a TypeError exception if the proto parameter isn't null or an object.

When I really want only key/value storage I usually use this pattern; also recently more and more often `new Map()`.

I use it when I don't want to create a child of `Object.protototype`, intentionally, to be able to use `key in dict` on objects that I use like maps i.e. only as data. Object is quite overloaded in JS and used for different cases, like class/instances, for inheritance, as the base for a lot of built in types (i.e. function) etc. 

I'd personally would treat this: `obj = {};` as a JavaScript Object and this: `dict = Object.create(null);` as a key/value storage where I know for sure that all keys it has are part of the information it carries.

I'd expect also, if `dict` inherits from another object used as a map object, that all its keys, also the ones inherited, are part of the data, I use this only seldom though.

With a `dict` I can savely use `for(key in dict)` and `if(key in dict)` without checking whether it is something coming from the prototype.
`Object.keys(dict)` would help here, but this and also `Object.hasOwnProperty` etc. are all unnecessary when you start with `Object.create(null)`; In a way these methods are workarounds (for too many uses of js objects) and they require me to write more code.

I'll probably stop this practice in most cases when `for key of mapInstance.keys()` is usable without setting special browser flags or pre-compiling via babel or whatever. However, I'm not convinced by your anti-pattern reasoning. I don't care about "writing C++/Java with JS syntax" arguments, the language is defined in that way.